### PR TITLE
Improve Intel FPGA Memory Attributes tests

### PR DESCRIPTION
**Pull request by mlychkov**
Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

**Comment by mlychkov** [comment]
@MrSidims @vmaksimo @AlexeySotkin could you please review?

**Comment by mlychkov** [comment]
All tests are failed on DebugInfo/COFF/no-cus.ll test that has nothing with my changes.

File: test/IntelFPGAMemoryAttributes.ll [hunk 1]
```diff
@@ -1,3 +1,180 @@
+; Source
```
Comment (AGindinson) [inline test/IntelFPGAMemoryAttributes.ll]: I'd suggest adding a link to the compiler repository & attaching the compilation command. If suitable, the same applies to `IntelFPGAMemoryAttributesForStruct.ll`

File: test/IntelFPGAMemoryAttributesForStruct.ll [hunk 2]
```diff
@@ -8,246 +169,476 @@
...
+%struct.doublepump_st = type { i32 }
+%struct.merge_st = type { i32 }
+%struct.max_replicates_st = type { i32 }
+%struct.templ_max_replicates_st = type { i32 }
+%struct.simple_dual_port_st = type { i32 }
+%struct.bank_bits_st = type { i32 }
+%struct.templ_bank_bits_st = type { i32 }
+%struct.state = type { [8 x i32] }
+
+; CHECK-LLVM: [[STR01:@[0-9_.]+]] = {{.*}}{memory:DEFAULT}{numbanks:4}
```
Comment (AGindinson) [inline test/IntelFPGAMemoryAttributesForStruct.ll]: Could `STR`/`FIELD` variables have some short-hand attribute names as suffixes instead of numbers?

**Comment by AlexeySotkin** [comment]
Please rebase on top of the master branch to re-run the tests.

File: test/IntelFPGAMemoryAttributes.ll (hunk 1 above)
Comment (mlychkov) [reply test/IntelFPGAMemoryAttributes.ll]: Done

File: test/IntelFPGAMemoryAttributesForStruct.ll (hunk 2 above)
Comment (mlychkov) [reply test/IntelFPGAMemoryAttributesForStruct.ll]: Done

**Comment by mlychkov** [comment]
> Could some motivation be added to the commit message? The changes themselves LGTM.

Commit message has been updated.

**Pull request merged**